core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java

```diff
@@ -336,6 +337,22 @@ void validate() {
         actionAfterProcess.equalsIgnoreCase("MOVE")) && Strings.isNullOrEmpty(targetFolder);
       Preconditions.checkArgument(!targetFolderEmpty, "Target folder cannot be empty for Action = '" +
         actionAfterProcess + "'.");
+
+      if (!Strings.isNullOrEmpty(pattern)) {
+        try {
+          Pattern.compile(pattern);
+        } catch (Exception e) {
+          throw new IllegalArgumentException(String.format("The regular expression '%s' is not valid.", pattern), e);
+        }
+        // By default, the Hadoop FileInputFormat won't return any files when using a regex pattern unless the path
+        // has globs in it. Checking for that scenario.
+        if (path.endsWith("/") ||
+          !(path.contains("*") || path.contains("?") || path.contains("{") || path.contains("["))) {
+          throw new IllegalArgumentException("When filtering with regular expressions, " +
+                                               "the Path must be a directory and leverage glob syntax. Most likely, " +
+                                               "you just need to end your folder path with a /*");
+        }
```

Comment:
throw new IllegalArgumentException("When filtering with regular expressions, " +
 +                                               "the path must be a directory and leverage glob syntax. Usually " +
 +                                               "the folder path needs to end with '/*'.");